autolab/Autolab
# Revamp Gradebook/Gradesheet to be more UI friendly

This PR aims to address some UI issues in the gradebook and gradesheet. Specifically for the gradesheet, @zackw brought up that there is no cursor to indicate that some columns can be sorted (wrong cursor and no icon), and that the tiny arrow is frustrating to try to click. The email can also be clicked to show further information about the student's submission, but there is also no indication that clicking the email will display a popover. There are lots of links, but they are not colored properly to indicate they can be clicked.

## Description


Current gradebook layout (lots of white-space, gradebook takes up barely any room, can't select text):
<img width="1510" alt="Screen Shot 2022-12-25 at 15 26 24" src="https://user-images.githubusercontent.com/25730111/209458927-f26cf388-ac12-4a40-8e72-48d33ab11cb3.png">

New gradebook layout (Use space more efficiently, move search to lefthand, selectable contents):
<img width="1512" alt="Screen Shot 2022-12-25 at 15 26 08" src="https://user-images.githubusercontent.com/25730111/209458935-c47624ad-b4a1-4ea6-b8e2-d4f116274fbb.png">

Current gradesheet layout:

<img width="1482" alt="Screen Shot 2022-12-25 at 15 29 19" src="https://user-images.githubusercontent.com/25730111/209458981-3ba74de1-ab4c-4d86-8cb7-8838d6eb01d9.png">

- No sorting icons / indicators

<img width="705" alt="Screen Shot 2022-12-25 at 15 33 19" src="https://user-images.githubusercontent.com/25730111/209459111-17ae8373-c814-488c-b1e5-bf852ef224db.png">

- popover is all grey, hard to know what's clickable

<img width="229" alt="Screen Shot 2022-12-25 at 15 33 34" src="https://user-images.githubusercontent.com/25730111/209459127-2ec4ee49-7d81-4f7d-ab7a-60cf397528df.png">

- tiny icon to view submission


New bradesheet layout:
<img width="1512" alt="Screen Shot 2022-12-25 at 15 28 59" src="https://user-images.githubusercontent.com/25730111/209458992-755d6f37-2635-4ab6-8c26-4d90ea8496a3.png">

- make wider to allow for more problems / content to be shown
- make gradesheet table scroll on overflow
- create actions column, have larger buttons to view / download submission
- combine lecture and section
- add sorting icons to sortable columns (up, down, sortable)
- change cursor to pointer on sortable columns

<img width="657" alt="Screen Shot 2022-12-25 at 15 34 54" src="https://user-images.githubusercontent.com/25730111/209459211-bfbdcb07-42c6-4039-82c0-77370db80462.png">

- Email is now blue, has pointer cursor to indicate it will show popover when clicked
- popover now has blue links

## How Has This Been Tested?




- Ensure gradebook functionality is the same (searching still works)
- Select text in gradebook

- Ensure searching, sorting works as expected in gradesheet, links to view and download submissions work
- If submission was filled in automatically, shouldn't be able to view/download submission
- clicking on email shows popover

## Types of changes


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
